lib/regular_expression/compiler/cranelift.rb

```diff
@@ -0,0 +1,384 @@
+# frozen_string_literal: true
+
+module RegularExpression
+  module Compiler
+    module Cranelift
+      class Compiled
+        attr_reader :f_ptr
+
+        def initialize(f_ptr)
+          @f_ptr = f_ptr
+        end
+
+        def to_proc
+          function = Fiddle::Function.new(@f_ptr, [Fiddle::TYPE_VOIDP, Fiddle::TYPE_SIZE_T], Fiddle::TYPE_SIZE_T)
+          lambda do |string|
+            value = function.call(string, string.length)
+            value if value != string.length + 1
+          end
+        end
+      end
+
+      def self.compile(cfg, _schedule)
```

Comment:
We should make the compiler request a schedule, rather than scheduling always done and then given to the compiler.